libtransmission/subprocess-posix.cc

```diff
@@ -82,35 +85,31 @@ void set_system_error(tr_error* error, int code, std::string_view what)
 
 [[nodiscard]] bool tr_spawn_async_in_parent(int pipe_fd, tr_error* error)
 {
-    int child_errno = 0;
-    ssize_t count = 0;
-
-    static_assert(sizeof(child_errno) == sizeof(errno));
-
-    do
+    for (;;)
     {
-        count = read(pipe_fd, &child_errno, sizeof(child_errno));
-    } while (count == -1 && errno == EINTR);
+        int child_errno = 0;
+        auto const n_read = read(pipe_fd, &child_errno, sizeof(child_errno));
 
-    close(pipe_fd);
```

Comment:
Looks like this PR is leaking file descriptors because this line is lost.

Reply:
Yikes, thanks for finding that!

 Regression fixed in f328c4c758b8a21b6b03a327a12c1d60745e8f3e